src/MuladdMacro.jl

```diff
@@ -87,23 +90,33 @@ end
 
 Determine whether `ex` is a dot call.
 """
-isdotcall(ex) =
-    (@capture(ex, (f_)(__)) && startswith(string(f), '.')) ||
-    @capture(ex, (_).(__))
+isdotcall(ex) = typeof(ex) <: Expr ? ex.head == :. || string(ex.args[1])[1] == '.' : false
```

Comment:
Does every expression always contain at least one argument? I always checked whether `isempty(ex.args)` since it seems `ex.args[1]` could throw an out-of-bound error?